integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/DatasetIdentifier.java

```diff
@@ -5,10 +5,24 @@
 
 package io.openlineage.spark.agent.util;
 
+import java.util.Optional;
 import lombok.Value;
 
 @Value
 public class DatasetIdentifier {
   String name;
   String namespace;
+  Optional<String> physicalLocation;
+
+  public DatasetIdentifier(String name, String namespace) {
+    this.name = name;
+    this.namespace = namespace;
+    physicalLocation = Optional.empty();
+  }
+
+  public DatasetIdentifier(String name, String namespace, Optional<String> physicalLocation) {
```

Comment:
`@Value` gives you this constructor

Reply:
We're using different variants of the construction through multiple places in code. I was not able to make Lombok generate different constructor variants and I did not wan't to change all the constructor calls to include the added physicalLocation. That's why I decided to go with multiple constructors put in the code.